components/feature/top-sites/src/test/java/mozilla/components/feature/top/sites/DefaultTopSitesStorageTest.kt

```diff
@@ -213,15 +215,15 @@ class DefaultTopSitesStorageTest {
 
         val frecentSite2 = TopFrecentSiteInfo("https://example.com", "Example")
         val frecentSite3 = TopFrecentSiteInfo("https://getpocket.com", "Pocket")
-        whenever(historyStorage.getTopFrecentSites(anyInt())).thenReturn(
+        whenever(historyStorage.getTopFrecentSites(anyInt(), any())).thenReturn(
```

Comment:
I really recommend doing away with the mocks, and just using the real storage implementation instead. It's a little bit more working setting things up, but really not much more - you'd need to actually insert history items and visits yourself to prep the storage.

For an example of how to set this up, see `browser-storage-sync` component. Specifically, you may need to copy-paste some bits from `build.gradle`:
```
// These dependencies are part of this module's public API.
    api(Dependencies.mozilla_places) {
        // Use our own version of the Glean dependency,
        // which might be different from the version declared by A-S.
        exclude group: 'org.mozilla.components', module: 'service-glean'
    }
...
testImplementation Dependencies.mozilla_places
testImplementation Dependencies.mozilla_full_megazord_forUnitTests
testImplementation Dependencies.mozilla_glean_forUnitTests
...
```

(it may work without the glean stuff, try that first; i don't recall the details right now)

The result should be a much more robust and useful test suite which exercises our whole stack! This will become a good integration test, instead of a "we're calling the right APIs" unit test. The latter is helpful, but the former actually gives you confidence that the system is working correctly.

Reply:
I spent some time on this, but ran into some issues that I have not managed to figure out yet. I will spin this task into a new issue https://github.com/mozilla-mobile/android-components/issues/9501 to not block this any further.